djlsp/parser.py

```diff
@@ -117,18 +117,32 @@ class DummyForLoop:
             )
         for variable, variable_type in self.context.items():
             if variable_type:
-                variable_import = ".".join(variable_type.split(".")[:-1])
-                script_lines.extend(
-                    [
-                        f"import {variable_import}",
-                        f"{variable}: {variable_type}",
-                    ]
-                )
+                variable_type_aliased = variable_type
+                # allow to use more complex types by splitting them into segments
+                # and try to import them separatly
+                for imp in set(filter(None, re.split(r"\[|\]| |,", variable_type))):
+                    variable_import = ".".join(imp.split(".")[:-1])
+                    if variable_import == "":
+                        continue
+
+                    # create import alias to allow variable to have same name as module
+                    import_alias = (
+                        "__" + hashlib.md5(variable_import.encode()).hexdigest()
+                    )
+                    variable_type_aliased = variable_type_aliased.replace(
+                        variable_import, import_alias
+                    )
+                    script_lines.append(f"import {variable_import} as {import_alias}")
+
+                script_lines.append(f"{variable}: {variable_type_aliased}")
             else:
                 script_lines.append(f"{variable} = None")
 
         # Add user code
-        script_lines.append(code)
+        # django uses abc.0 for list index lookup, replace those with abc[0]
+        script_lines.append(re.sub(r"(.*?)\.(\d+)", r"\1[\2]", code))
```

Comment:
Could this not be simpler by `re.sub(r"\.(\d+)", r"[\1]", code)`

Reply:
Youre right, have updated it.